AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/nativead/TemplateNativeAdActivity.java

```diff
@@ -0,0 +1,114 @@
+package com.applovin.enterprise.apps.demoapp.ads.max.nativead;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.FrameLayout;
+
+import com.adjust.sdk.Adjust;
+import com.adjust.sdk.AdjustAdRevenue;
+import com.adjust.sdk.AdjustConfig;
+import com.applovin.enterprise.apps.demoapp.R;
+import com.applovin.enterprise.apps.demoapp.ui.BaseAdActivity;
+import com.applovin.mediation.MaxAd;
+import com.applovin.mediation.MaxAdRevenueListener;
+import com.applovin.mediation.MaxError;
+import com.applovin.mediation.nativeAds.MaxNativeAdListener;
+import com.applovin.mediation.nativeAds.MaxNativeAdLoader;
+import com.applovin.mediation.nativeAds.MaxNativeAdView;
+
+public class TemplateNativeAdActivity
+        extends BaseAdActivity
+        implements MaxAdRevenueListener
+{
+    // Map of ad unit IDs to native ad loaders
+    private MaxNativeAdLoader nativeAdLoader;
+    private MaxAd             nativeAd;
+
+    private FrameLayout nativeAdLayout;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate( savedInstanceState );
+        setContentView( R.layout.activity_native_template );
+        setTitle( R.string.activity_template_natives );
+
+        nativeAdLayout = findViewById( R.id.native_ad_layout );
+        setupCallbacksRecyclerView();
+
+        String adUnitId = "YOUR_AD_UNIT_ID";
+
+        nativeAdLoader = new MaxNativeAdLoader( adUnitId, this );
+        nativeAdLoader.setPlacement( "Native Custom Test Placement" );
+        nativeAdLoader.setExtraParameter( "test_extra_key", "test_extra_value" );
+        nativeAdLoader.setRevenueListener( this );
+        nativeAdLoader.setNativeAdListener( new MaxNativeAdListener()
+        {
+            @Override
+            public void onNativeAdLoaded(final MaxNativeAdView nativeAdView, final MaxAd ad)
+            {
+                logCallback( 2 );
```

Comment:
No idea what passing in `2` means, not very idiomatic. I get what you're doing though, so maybe you can just create new function `logAnonymousecallback(...)` or something.